includes/database/class-wp-db-table.php

```diff
@@ -0,0 +1,340 @@
+
+		// Bail if no upgrade needed
+		if ( version_compare( (int) $this->db_version, (int) $this->version, '>=' ) ) {
+			return;
+		}
+
+		// Bail if global and upgrading global tables is not allowed
+		if ( $this->is_global() && ! wp_should_upgrade_global_tables() ) {
+			return;
+		}
+
+		// Create or upgrade?
+		$this->exists()
+			? $this->upgrade()
+			: $this->create();
+
+		// Only set database version if table exists
+		if ( $this->exists() ) {
+			$this->set_db_version();
+		}
+	}
+
+	/** Private ***************************************************************/
+
+	/**
+	 * Setup the necessary table variables
+	 *
+	 * @since 1.1.0
+	 */
+	private function setup() {
+
+		// Setup database
+		$this->db   = $GLOBALS['wpdb'];
+		$this->name = sanitize_key( $this->name );
+
+		// Maybe create database key
+		if ( empty( $this->db_version_key ) ) {
+			$this->db_version_key = "wpdb_{$this->name}_version";
+		}
+	}
+
+	/**
+	 * Modify the database object and add the table to it
+	 *
+	 * This must be done directly because WordPress does not have a mechanism
+	 * for manipulating them safely
+	 *
+	 * @since 1.1.0
+	 */
+	private function set_wpdb_tables() {
+
+		// Global
+		if ( $this->is_global() ) {
+			$prefix                       = $this->db->get_blog_prefix( 0 );
+			$this->db->{$this->name}      = "{$prefix}{$this->name}";
+			$this->db->ms_global_tables[] = $this->name;
+
+		// Site
+		} else {
+			$prefix                  = $this->db->get_blog_prefix( null );
+			$this->db->{$this->name} = "{$prefix}{$this->name}";
+			$this->db->tables[]      = $this->name;
+		}
+
+		// Set the table name locally
+		$this->table_name = $this->db->{$this->name};
+
+		// Charset
+		if ( ! empty( $this->db->charset ) ) {
+			$this->charset_collation = "DEFAULT CHARACTER SET {$this->db->charset}";
+		}
+
+		// Collation
+		if ( ! empty( $this->db->collate ) ) {
+			$this->charset_collation .= " COLLATE {$this->db->collate}";
+		}
+	}
+
+	/**
+	 * Set the database version for the table
+	 *
+	 * Global table version in "_sitemeta" on the main network
+	 *
+	 * @since 1.1.0
+	 */
+	private function set_db_version() {
+
+		// Set the class version
+		$this->db_version = $this->version;
+
+		// Update the DB version
+		$this->is_global()
+			? update_network_option( null, $this->db_version_key, $this->version )
+			:         update_option(       $this->db_version_key, $this->version );
+	}
+
+	/**
+	 * Get the table version from the database
+	 *
+	 * Global table version from "_sitemeta" on the main network
+	 *
+	 * @since 1.1.0
+	 */
+	private function get_db_version() {
+		$this->db_version = $this->is_global()
+			? get_network_option( null, $this->db_version_key, false )
+			:         get_option(       $this->db_version_key, false );
+	}
+
+	/**
+	 * Add class hooks to WordPress actions
+	 *
+	 * @since 1.1.0
+	 */
+	private function add_hooks() {
+
+		// Activation hook
+		register_activation_hook( __FILE__, array( $this, 'maybe_upgrade' ) );
```

Comment:
I don't think this will work since this file doesn't get activated.

Reply:
Fixed in c7b25eeafb48ad0123af6546759398aa3d0cb7f9.